src/Transform/ONNX/ONNXOpTransformPass.cpp

```diff
@@ -59,12 +58,20 @@ struct ONNXOpTransformPass : public mlir::PassWrapper<ONNXOpTransformPass,
       llvm::cl::desc("max iteration for op transform passes."),
       llvm::cl::init(3)};
 
+  Option<bool> onnxOpTransformReport{*this, "onnx-op-transform-report",
+      llvm::cl::desc("Report diagnostic info for op transform passes."),
+      llvm::cl::init(false)};
+
   ONNXOpTransformPass() = default;
   ONNXOpTransformPass(const ONNXOpTransformPass &pass)
       : mlir::PassWrapper<ONNXOpTransformPass,
-            OperationPass<mlir::ModuleOp>>() {}
-  ONNXOpTransformPass(int threshold_) {
-    this->onnxOpTransformThreshold = threshold_;
+            OperationPass<mlir::ModuleOp>>() {
+    this->onnxOpTransformThreshold = pass.onnxOpTransformThreshold;
+    this->onnxOpTransformReport = pass.onnxOpTransformReport;
+  }
+  ONNXOpTransformPass(int threshold, bool report) {
+    this->onnxOpTransformThreshold = threshold;
+    this->onnxOpTransformReport = report;
```

Comment:
Prefer initializing member variables using constructors init list

Reply:
Unfortunately, i cannot initialize Pass objects in this fashion as they're copy-constructor has been deleted. An assignment is required to initialize them in the way we want.